src/telemetry/mod.rs

```diff
@@ -0,0 +1,123 @@
+pub(crate) mod metrics;
+
+use itertools::Itertools;
+use once_cell::sync::OnceCell;
+use opentelemetry::sdk::trace::Config;
+use opentelemetry::sdk::Resource;
+use opentelemetry::{global, sdk::metrics::PushController, util::tokio_interval_stream, KeyValue};
+use opentelemetry_otlp::WithExportConfig;
+use std::collections::VecDeque;
+use std::time::Duration;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
+
+const TELEM_SERVICE_NAME: &str = "temporal-core-sdk";
+const ENABLE_OPENTELEM_ENV_VAR: &str = "TEMPORAL_ENABLE_OPENTELEMETRY";
```

Comment:
Let's avoid requiring env vars to configure core, maybe only run this with config=test?